lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1053,6 +1059,14 @@ syncProgress epochLength tip slotNow =
         Quantity $ toEnum $ fromIntegral $
             (100 * bhTip) `div` (bhTip + n1 - n0)
 
+syncProgressRelativeToCurrentTime
+    :: SlotParameters
+    -> BlockHeader
+    -> IO (Quantity "percent" Percentage)
```

Comment:
:scream: ... `IO` in the `Primitive.Types` module!

More seriously, the time should really be a parameter here. Let's not have effects down here. We've kept it pure until now :) ... Also, shouldn't this be in another PR?

Aaaand.... we've lost the comment that was giving a justification for this unsafe pattern matching :/ ... That's kinda bad.